redis/commands/core.py

```diff
@@ -1725,7 +1745,19 @@ def get(self, name: KeyT) -> ResponseT:
 
         For more information see https://redis.io/commands/get
         """
-        return self.execute_command("GET", name)
+        if not self._is_pipeline and self._cache_write_once:
+            name = to_str(name)
+            key_type = get_key_type(name, self._key_types)
+            if key_type == KeyCacheProp.WRITE_ONCE:
+                val = self._key_cache.get(name, None)
+                if val is None:
+                    val = self.execute_command("GET", name)
+                    self._key_cache[name] = val
+                return val
+            else:
+                return self.execute_command("GET", name)
+        else:
+            return self.execute_command("GET", name)
```

Comment:
This code is really copy-pasted everywhere. I think it is better to do it in a Python decorator:

```
@popper
def get(..) -> ResponseT:
```

And the `@popper` decorator does all these checks. You can read about decorators here: https://www.geeksforgeeks.org/decorators-in-python/